doc/src/manual/documentation.md

```diff
@@ -81,7 +81,25 @@ As in the example above, we recommend following some simple conventions when wri
    ...
    """
    ```
-5. Include any code examples in an `# Examples` section.
+5. Provide hints to similar functions
+   
+   Sometimes there are functions of similar functionality, which are named quite different. To increase 
+   discoverability please provide a short list of these similar functions in a `See also: ` section.
+   
+   ```
+   See also: [`bar!`](@ref), [`baz`](@ref), [`baaz`](@ref)
```

Comment:
Maybe this should be a `#` section, just like `# Examples`?

Reply:
I think a heading is a bit too verbose. `# See also: pop!, shift!` renders as
```julia
     See also: pop!, shift!
    ≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡≡
```

I would rather bold `**See also:**` to add visibility.

I know, i called it `section` in the docs, which might be a bit confusing, if it's not rendered as a section...

Reply:
Having it like this is fine and stuff like making these into sections should probably be done all at once.

Reply:
I think there's a more general issue (even for `# Examples`) with the three lines which are too heavy (see https://github.com/JuliaLang/julia/issues/22870). So I'd rather use a consistent style with one section for "See also " and one section for "Examples", and fix both at the same time later. Note that the REPL isn't the only display format: there's also the HTML manual, where sections look much nicer (see screenshot at #22870).

Also, since you recommend to use a bullet list when describing functions, it's important to clearly separate it from the description of the function itself. So overall I suggest always having `# See also` section heading just like `# Examples`, and then always a line break, followed either by a list of functions on a single line, or list with bullet points.

Reply:
I don't think it should be a section like that, I think it could even just be the last sentence of the docstring:
```
    push!(collection, items...) -> collection

Insert one or more items at the end of collection. See also: `append!`, `pop!`

# Examples
[...]
```

IMO this should just be a tiny side note, definitely not as important as examples for the function.

Reply:
OK, I see you dropped the suggestion to use a bullet list. Let's go with the lighter solution then, but at least add a line break before it. Also please remove the term "section"  (maybe call it "paragraph").